### ISSUE TITLE
Feature Request: Add function New-CertificateSigningRequestEx

### ISSUE THREAD

**Issue by PatrickOnGit** [Crypt32/PSPKI#78]
The function New-SelfSignedCertificateEx does not allow for creating CSRs. Neither does the SysadminsLV.PKI.Cryptography.X509Certificate.X509CertificateBuilder class.
So it would be great to have this function.

**Comment by Crypt32** [comment]
Next two months I will be very busy, but I will put this in my ToDo list.

I think, there will be two versions:
1) cross-platform with PKCS#1/PKCS#8 key storage (similar to OpenSSL). I know, there is `CertificateRequest` class in .NET 4.7.2  and .NET core, but I think, it is not flexible.
2) Microsoft-specific that will utilize windows certificate store.

If you have other ideas, drop them in this thread.

**Comment by PatrickOnGit** [comment]
Hello Vadims

This sounds great. Thank you. 
For PKCS#1/PKCS#8 key storage you might allow for encrypted and unencrypted PEM-key.

What I'm also looking for is an option to sign CSR with a RA Agent certificate.

Using PowerShells capability for dynamic parameters, it would be kind of cool if the commands would allow for tab completion for available Provider and Algorithm types etc. Your command Get-CryptographicServiceProvider already reveals this information. 

Thank you
Kind regards
Patrick

________________________________
Von: Vadims Podans <notifications@github.com>
Gesendet: Mittwoch, 1. Mai 2019 15:24
An: Crypt32/PSPKI
Cc: PatrickOnGit; Author
Betreff: Re: [Crypt32/PSPKI] Feature Request: Add function New-CertificateSigningRequestEx (#78)

Next two months I will be very busy, but I will put this in my ToDo list.

I think, there will be two versions:

cross-platform with PKCS#1/PKCS#8 key storage (similar to OpenSSL). I know, there is CertificateRequest class in .NET 4.7.2 and .NET core, but I think, it is not flexible.
Microsoft-specific that will utilize windows certificate store.

If you have other ideas, drop them in this thread.

**Comment by PatrickOnGit** [comment]
is there a plan to work on this request?

**Comment by Crypt32** [comment]
Yes, I'm considering this as well. In next version, I'm going to revisit existing GitHub items and work on them.

**Comment by PatrickOnGit** [comment]
thank you
